src/librustc_codegen_ssa/back/link.rs

```diff
@@ -1843,3 +1843,55 @@ fn is_pic(sess: &Session) -> bool {
 
     reloc_model_arg == "pic"
 }
+
+pub fn find_crate_name(sess: Option<&Session>, attrs: &[ast::Attribute], input: &Input) -> String {
+    let validate = |s: String, span: Option<Span>| {
+        rustc_metadata::validate_crate_name(sess, &s, span);
```

Comment:
Why is there a function with that name in *`rustc_metadata`*?!

Reply:
It definitely looks like it should be in `rustc_session`: https://github.com/rust-lang/rust/blob/5a72ecf2c5c97933cecce91cb58d104da8120624/src/librustc_metadata/lib.rs#L33-L65

Reply:
Heh, it used to be in `librustc`: https://github.com/rust-lang/rust/commit/d4beecaed36f6c8ba8f73f06d89fd20267887749

Reply:
Moved them both to rustc_session::output